### PR TITLE
Use TMC infra to run E2E CI jobs.

### PR DESCRIPTION
**Pull request by HanFa**
**What this PR does / why we need it**:

This PR connects the [test-infra definition of E2E CI job](https://github.com/kubernetes/test-infra/blob/779a78208956a3aa07b77b81980dab2906649099/config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml#L252) to the [E2E entry point](https://github.com/kubernetes/cloud-provider-vsphere/blob/master/hack/e2e.sh) `hack/e2e.sh` defined in this project. It allows upstream CPV to leverage TMC infrastructure for E2E jobs. So far, the E2E can successfully create a bootstrap cluster inside prow jenkins container, and a workload cluster on TMC vSphere.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

Future tests will be based on this PR.

**Release note**:

```release-note
Use TMC infra to run E2E CI jobs.
```

### CODE REVIEW

File: hack/e2e.sh [hunk 1]
```diff
@@ -47,8 +50,47 @@ function login() {
...
+  gcr.io/cluster-api-provider-vsphere/extra/openvpn:latest
+
+# Tail the vpn logs
+docker logs vpn
+
+# Sleep to allow vpn container to start running
+sleep 30
+
+# Retrieve an IP to be used as the kube-vip IP
+KUBECONFIG="/root/ipam-conf/capv-services.conf"
```
Comment (lubronzhan) [inline hack/e2e.sh:68]: Who generate this file？

**Comment by lubronzhan** [comment]
Currently, it doesn't install CPI I assume?

File: hack/e2e.sh (hunk 1 above)
Comment (HanFa) [reply hack/e2e.sh:68]: it's mounted when using this `preset-cluster-api-provider-vsphere-e2e-config` preset https://github.com/kubernetes/test-infra/blob/28f5d976bee58965855f1b2ae9c0a9a9dd931100/config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-presubmits.yaml#L31

**Comment by HanFa** [comment]
> Currently, it doesn't install CPI I assume?

No, it doesn't. That is the next step.

**Comment by lubronzhan** [comment]
/lgtm
/approve
great!

**Pull request merged**